src/ert/scheduler/local_driver.py

```diff
@@ -66,18 +68,19 @@ async def _init(
             preexec_fn=os.setpgrp,
         )
 
-    async def _wait(self, proc: Process) -> bool:
+    async def _wait(self, proc: Process) -> int:
         """This method exists to allow for mocking it in tests"""
-        return await proc.wait() == 0
+        return await proc.wait()
 
-    async def _kill(self, proc: Process) -> None:
+    async def _kill(self, proc: Process) -> int:
         """This method exists to allow for mocking it in tests"""
         try:
             proc.terminate()
             await asyncio.wait_for(proc.wait(), _TERMINATE_TIMEOUT)
         except asyncio.TimeoutError:
             proc.kill()
             await asyncio.wait_for(proc.wait(), _TERMINATE_TIMEOUT)
+        return proc.returncode or -1
```

Comment:
so if `returncode == 0` we should return -1 ? (or did you mean `is None`?)

Reply:
We should return non-zero is my thinking. Otherwise we can end up with `FinishedEvent(iens=iens, returncode=0, aborted=True)`, or:

![Task failed successfully](https://www.mytechblog.net/wp-content/uploads/2021/03/1-ON_d7DWgW8g8uu3EBntfNw.png)